b2share/factory.py

```diff
@@ -1,192 +1,178 @@
-    rest_app -> schemas_views [label="endpoint in\\n['/api/communities/<ID>/schemas/*', ...]"]
-    rest_app -> other_views [label="endpoint == '...'"]
-    }
-
-See Invenio and invenio_base module for more information regarding how Invenio
-applications are created. The *"UI Application"* in our case is a custom
-one, it does not match *"Invenio UI application"* which serves default
-Invenio UI.
-
-The ``*\*.views*`` are the ``views.py`` modules included in modules which
-contain the REST API definition. The requests are dispatched to the right
-view class using the Flask endpoints matching rules.
-
-**TODO**: Note that Invenio has evolved since :py:func:`~.create_app` was
-created. It will be necessary at some point to refactor it.
-"""
+# Invenio is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Flask application factories for Invenio flavours."""
+
+from __future__ import absolute_import, print_function
 
 import os
 import sys
 
-from flask import Flask
+import pkg_resources
+
+from flask import current_app
+
 from invenio_base.app import create_app_factory
-from invenio_config import create_conf_loader
-from werkzeug.wsgi import DispatcherMiddleware
-from werkzeug.contrib.fixers import ProxyFix
+from invenio_base.wsgi import create_wsgi_factory, wsgi_proxyfix
+from invenio_base.signals import app_created, app_loaded
+from invenio_cache import BytecodeCache
+from invenio_config import create_config_loader
+
+from jinja2 import ChoiceLoader, FileSystemLoader
 
 from . import config
 
 env_prefix = 'B2SHARE'
 
-config_loader = create_conf_loader(config=config, env_prefix=env_prefix)
+invenio_config_loader = create_config_loader(
+    config=config, env_prefix=env_prefix
+)
+
+@app_created.connect
+def receiver_app_created(sender, app=None, **kwargs):
+    app.logger.debug("Application created")
+
+@app_loaded.connect
+def receiver_app_loaded(sender, app=None, **kwargs):
+    app.logger.debug("Application Loaded")
+    app.logger.debug("Instance Path: {}".format(app.instance_path))
+    check_configuration(app.config, app.logger)
+
+def instance_path():
+    """Instance path for Invenio.
+
+    Defaults to ``<env_prefix>_INSTANCE_PATH``
+     or if environment variable is not set ``<sys.prefix>/var/instance``.
+    """
+    return os.getenv(env_prefix + '_INSTANCE_PATH') or \
+        os.path.join(sys.prefix, 'var', 'instance')
+
+
+def static_folder():
+    """Static folder path.
 
-instance_path = os.getenv(env_prefix + '_INSTANCE_PATH') or \
-    os.path.join(sys.prefix, 'var', 'b2share-instance')
-"""Instance path for B2Share."""
+    Defaults to ``<env_prefix>_STATIC_FOLDER``
+    or if environment variable is not set ``<sys.prefix>/var/instance/static``.
+    """
+    return os.getenv(env_prefix + '_STATIC_FOLDER') or \
+        os.path.join(instance_path(), 'static')
```

Comment:
V2 codebase uses B2SHARE_UI_PATH.
https://github.com/EUDAT-B2SHARE/b2share/blob/d846068c71853221ac53239c9efc56e3e795e6c3/b2share/factory.py#L133-L135

** Might need to be changed, or not**